spec/controllers/submissions_controller_spec.rb

```diff
@@ -0,0 +1,638 @@
+        @submission_collector.expects(:push_groupings_to_queue).once
+        get_as @admin, :collect_all_submissions, assignment_id: 1, id: 1
+        expect(flash[:success]).to eq(
+         I18n.t('collect_submissions.collection_job_started',
+                assignment_identifier: 'a1'))
+        is_expected.to respond_with(:redirect)
+      end
+
+      it 'should be able to release submissions' do
+        allow(Assignment).to receive(:find) {@assignment}
+        post_as @admin,
+                :update_submissions,
+                assignment_id: 1,
+                groupings: ([] << @assignment.groupings).flatten,
+                release_results: 'true'
+        is_expected.to respond_with(:success)
+      end
+    end
+
+    it 'download all files uploaded in a Zip file' do
+      @file1_name = 'TestFile.java'
+      @file2_name = 'SecondFile.go'
+      @file1_content = "Some contents for TestFile.java\n"
+      @file2_content = "Some contents for SecondFile.go\n"
+
+      @group.access_repo do |repo|
+        txn = repo.get_transaction('test')
+        path = File.join(@assignment.repository_folder, @file1_name)
+        txn.add(path, @file1_content, '')
+        path = File.join(@assignment.repository_folder, @file2_name)
+        txn.add(path, @file2_content, '')
+        repo.commit(txn)
+
+        # Generate submission
+        @submission = Submission.generate_new_submission(
+                       @grouping,
+                       repo.get_latest_revision)
+      end
+      get_as @admin,
+             :downloads,
+             assignment_id: @assignment.id,
+             id: @submission.id,
+             grouping_id: @grouping.id
+
+      expect('application/zip').to eq(response.header['Content-Type'])
+      is_expected.to respond_with(:success)
+      zip_path = "tmp/#{@assignment.short_identifier}_" +
+                 "#{@grouping.group.group_name}_r#{@grouping.group.repo
+                     .get_latest_revision.revision_number}.zip"
+      Zip::File.open(zip_path) do |zip_file|
+        file1_path = File.join("#{@assignment.repository_folder}-" +
+                                   "#{@grouping.group.repo_name}",
+                               @file1_name)
+        file2_path = File.join("#{@assignment.repository_folder}-" +
+                                   "#{@grouping.group.repo_name}",
+                               @file2_name)
+        expect(zip_file.find_entry(file1_path)).to_not be_nil
+        expect(zip_file.find_entry(file2_path)).to_not be_nil
+
+        expect(zip_file.read(file1_path)).to eq(@file1_content)
+        expect(zip_file.read(file2_path)).to eq(@file2_content)
+      end
+    end
+
+    it 'not be able to download an empty revision' do
+      @group.access_repo do |repo|
+        txn = repo.get_transaction('test')
+        repo.commit(txn)
+
+        # Generate submission
+        @submission = Submission.generate_new_submission(
+                          @grouping, 
```

Comment:
Indent the first parameter one step more than the previous line.<br>Trailing whitespace detected.